File: src/main/java/io/github/slimefunguguproject/bump/implementation/groups/AppraiseInfoMenu.java
package io.github.slimefunguguproject.bump.implementation.groups;

import java.util.List;
import java.util.function.Consumer;

import javax.annotation.Nonnull;
import javax.annotation.ParametersAreNonnullByDefault;

import org.bukkit.Material;
import org.bukkit.inventory.ItemStack;

import io.github.bakedlibs.dough.items.CustomItemStack;
import io.github.slimefunguguproject.bump.api.appraise.AppraiseType;
import io.github.slimefunguguproject.bump.implementation.Bump;
import io.github.slimefunguguproject.bump.implementation.menus.AppraiseTypesMenu;
import io.github.slimefunguguproject.bump.utils.AppraiseUtils;

final class AppraiseInfoMenu extends AppraiseTypesMenu {
    @ParametersAreNonnullByDefault
    public AppraiseInfoMenu(String name, Consumer<AppraiseType> successCallback, Runnable backCallback) {
        super(name, successCallback, backCallback);
    }

    @Nonnull
    @Override
    public ItemStack getDisplayItem(@Nonnull AppraiseType type) {
        List<String> lore = AppraiseUtils.getDescriptionLore(type);
        lore.add("");
        lore.add(Bump.getLocalization().getString("appraise_info.click"));

        return new CustomItemStack(
            Material.PAPER,
            Bump.getLocalization().getString("appraise_info.name", type.getName()),
            lore
        );
    }
}

File: src/main/kotlin/io/github/slimefunguguproject/bump/implementation/groups/AppraiseInfoMenu.kt
package io.github.slimefunguguproject.bump.implementation.groups

import io.github.slimefunguguproject.bump.Bump
import io.github.slimefunguguproject.bump.api.appraise.AppraiseType
import io.github.slimefunguguproject.bump.implementation.menu.AppraiseTypesMenu
import io.github.slimefunguguproject.bump.utils.items.MaterialType
import org.bukkit.Material
import java.util.function.Consumer

internal class AppraiseInfoMenu(
    name: String,
    successCallback: Consumer<AppraiseType>,
    backCallback: Runnable
) : AppraiseTypesMenu(name, successCallback, backCallback) {
    override fun getInfoItem() =
        Bump.localization.getGuiItem(MaterialType.Material(Material.NAME_TAG), "APPRAISE_MENU_INFO")
}
